Code/Framework/AzFramework/AzFramework/Spawnable/SpawnableEntitiesManager.cpp

```diff
@@ -469,6 +503,40 @@ namespace AzFramework
         }
     }
 
+    bool SpawnableEntitiesManager::ProcessRequest(DespawnEntityCommand& request)
+    {
+        Ticket& ticket = *request.m_ticket;
+        if (request.m_requestId == ticket.m_currentRequestId)
+        {
+            AZStd::vector<AZ::Entity*>& spawnedEntities = request.m_ticket->m_spawnedEntities;
+            for (auto entityIterator = spawnedEntities.begin(); entityIterator != spawnedEntities.end(); ++entityIterator)
+            {
+                if (*entityIterator != nullptr && (*entityIterator)->GetId() == request.m_entityId)
+                {
+                    // Setting it to 0 is needed to avoid the infite loop between GameEntityContext and SpawnableEntitiesManager.
+                    (*entityIterator)->SetSpawnTicketId(0);
+                    GameEntityContextRequestBus::Broadcast(
+                        &GameEntityContextRequestBus::Events::DestroyGameEntity, (*entityIterator)->GetId());
+                    AZStd::iter_swap(entityIterator, spawnedEntities.rbegin());
```

Comment:
nit: This might be clearer as the erase-remove idiom, e.g. `spawnedEntities.erase(AZStd::remove(entityIterator));`

Reply:
Ah yes omitting iterators will forever be my bane, heh - yeah if we're using the other idiom for a pop and swap everywhere else that's fine

Reply:
The syntax @srikappa-amzn wrote is correct. It would however perform slightly worse than a normal `erase` on a vector. This is because the normal `erase` removes an element and moves all proceeding elements down a slot. With the `remove` it needs to keep the element around to move to the end of the vector and then it gets removed.
That said, the swap and pop is significantly faster for containers that don't need to maintain their relative position because it only requires a single swap and `pop_back` which is cheaper than an erase. (The above can be made even faster with `*entityIterator = AZStd::,move(spawnedEntities.back());` but that's minor.)